.circleci/config.yml

```diff
@@ -4,51 +4,105 @@
 #
 version: 2
 jobs:
-  deploy:
+  deploy-dev:
     docker:
-    - image: circleci/node:10.12.0
+      - image: circleci/node:10.12.0
     parallelism: 1
     working_directory: ~/repo
     steps:
-    - checkout
-    - restore_cache:
-        name: Restore Yarn Package Cache
-        keys:
-        - yarn-packages-{{ checksum "yarn.lock" }}
-    - run:
-        name: Install Dependencies
-        command: yarn install --frozen-lockfile --cache-folder ~/.cache/yarn
-    - save_cache:
-        name: Save Yarn Package Cache
-        key: yarn-packages-{{ checksum "yarn.lock" }}
-        paths:
-        - ~/.cache/yarn
-    - run: yarn build
-    - run:
-        name: Deploy to hosting
-        # Package dist folder
-        # Do some hacks to be able to copy file without ssh key (only with user/password). That's not needed for normal hosting
-        # Copy package to tmp folder
-        # On remote host:
-        # - remove public folder contents
-        # - unpack bundle archive into public folder
-        # - remove bundle archive
-        command: |
-          tar -czf dist.tar.gz ~/repo/dist -C ~/repo/dist .
-          sudo apt install sshpass
-          ssh-keyscan -t rsa $DEPLOY_URL >> ~/.ssh/known_hosts
-          sshpass -p $DEPLOY_PASSWORD scp ~/repo/dist.tar.gz $DEPLOY_USER@$DEPLOY_URL:~/tmp
-          sshpass -p $DEPLOY_PASSWORD ssh $DEPLOY_USER@$DEPLOY_URL /bin/bash << EOF
-            rm -rf $DEPLOY_PATH/*
-            tar -xzf ~/tmp/dist.tar.gz -C $DEPLOY_PATH
-            rm -f ~/tmp/dist.tar.gz
-          EOF
+      - checkout
+      - restore_cache:
```

Comment:
There is too much dublication going on.
You can split processes into multiple jobs

```yaml
 build-staging:
    jobs:
      - build
      - test
      - ship-to-staging-server

 build-production:
    jobs:
      - appy-production-env-variables
      - build
      - test
      - ship-to-production server
```

Reply:
Good point! I've added aliases